sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnostics.java

```diff
@@ -100,7 +100,7 @@ public CosmosDiagnosticsContext getDiagnosticsContext() {
     }
 
     void setDiagnosticsContext(CosmosDiagnosticsContext ctx) {
-        checkNotNull("ctx", "Argument 'ctx' must not be null.");
+        checkNotNull(ctx, "Argument 'ctx' must not be null.");
```

Comment:
Wasn't checking the input, was instead validating a string constant.

Reply:
great catch, thanks!